app/src/main/java/org/hackillinois/android/notifications/HackIllinoisNotificationManager.kt

```diff
@@ -0,0 +1,81 @@
+package org.hackillinois.android.notifications
+
+import android.app.*
+import android.content.Context
+import android.content.Intent
+import android.graphics.BitmapFactory
+import android.os.Build
+import org.hackillinois.android.R
+import org.hackillinois.android.database.entity.Event
+import java.util.*
+
+class HackIllinoisNotificationManager {
+    companion object {
+        private val MINUTES_BEFORE_TO_NOTIFY = 10
+        private val MILLIS_IN_MINUTE = 60 * 1000
+        private val REQUEST_CODE = 999999
+
+        private val CHANNEL_ID = "event_notification_channel"
+        private val CHANNEL_NAME = "HackIllinois"
+
+        fun scheduleEventNotification(context: Context, event: Event) {
+            val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+            val notification = buildEventNotification(context, event)
+            val alarmIntent = buildNotificationPendingIntent(context, notification)
+            val timeToTrigger = event.getStartTimeMs() - MINUTES_BEFORE_TO_NOTIFY * MILLIS_IN_MINUTE
+
+            // don't set an alarm for an event that has already occurred
```

Comment:
There are a lot of negatives here... we could just do `if time <= event.start then alarm.set()`